test/plugin/secretless_test_harness/main.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"log"
+	"net"
+	"os"
+
+	secretlessLog "github.com/cyberark/secretless-broker/internal/log"
+	"github.com/cyberark/secretless-broker/internal/plugin"
+	tcpinternal "github.com/cyberark/secretless-broker/internal/proxyservice/tcp"
+)
+
+// ListenPort is the port that the test harness TCP proxy service is listening on
+const ListenPort = "6175"
+
+// This test harness is for validating plugins against Secretless by passing the plugin
+// as a connector to a TCP proxy service. It avoids the overhead of creating `net.Listener`s
+// and ProxyService instances from Secretless Config. Instead, it creates a single TCP
+// proxy service and the required net.Listener.
+func main() {
+	logger := secretlessLog.New(true)
+
+	logger.Infof("listening on %s\n", ListenPort)
+
+	// create the net.Listener that will be used by the test harness TCP service
+	// proxy
+	server, err := net.Listen("tcp", "localhost:"+ListenPort)
+	if err != nil {
+		log.Printf("error: %v\n", err)
+		os.Exit(1)
+	}
+
+	defer server.Close()
+
+	// get all plugins
+	plugins, err := plugin.AllAvailablePlugins(
+		"../example/plugin",
+		"",
+		logger,
+		)
+	if err != nil {
+		log.Printf("error: %v\n", err)
+		os.Exit(1)
+	}
+
+	tcpPlugins := plugins.TCPPlugins()
+
+	// only use first TCP plugin
+	var pluginId string
+	for k := range tcpPlugins {
```

Comment:
What's `k` in this context? I get that there's a bit of a preference for single-letter vars but this one I can't even tell what it's trying to convey.

Reply:
Oh yeah. I thought the comment would be helpful in explaining what's happening i.e. Extracting the first TCP plugin

Reply:
Hmm ... I guess I get that you're extracting the first plugin but `k` is neither short for `plugin` nor `first` so it's hard to extract its relevant meaning in this loop without reading the comment and the logic of how its used. Can we do `p` to make it obvious we're iterating over plugins?